### PR TITLE
fix(): request & response body handling

### LINKED ISSUE #84: Requests with body not in JSON format failing
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Follow-up to https://github.com/nestjs/azure-func-http/issues/78.

Any request providing body that is not JSON (see #78) is failing (POST, PUT) due to the fact that express middleware expect an open stream for the request, and currently it received an event emitter through `azure-function-express` wrapper.

## Expected behavior
Any kind of valid request should be processable like a regular NestJS app.

## Minimal reproduction of the problem with instructions
See https://github.com/nestjs/azure-func-http/issues/78

## What is the motivation / use case for changing the behavior?

Support file upload, form-encoded data...

## Environment
<pre><code>
Nest version: 6.12.2

 
For Tooling issues:
- Node version: 12.13.0
- Platform:  Mac

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by sinedied**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Fix #84, and same issue when using native Nest router.

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?
Use `azure-function-express` to wrap req/res

Issue Number: #84 #89 

## What is the new behavior?
Use same req/res wrappers in all adapters (express/nest-http).

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```
There's no surfaced breaking change, but since `azure-function-express` has been removed the adaptor behavior may behave differently for some middleware (in all my tested use cases, no issues so far).

## Other information

### CODE REVIEW

**Comment by sinedied** [comment]
Added also the fix for #89.

I hope this can be merged soon, let me know if something's unclear of it needs improvements 😄

File: package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
   "dependencies": {
     "@angular-devkit/schematics": "8.3.22",
     "@schematics/angular": "8.3.22",
-    "azure-function-express": "2.0.0",
     "cors": "2.8.5",
+    "reflect-metadata": "^0.1.13",
```
Comment (kamilmysliwiec) [inline package.json]: `reflect-metadata` should not be defined as a dependency (multiple copies of the `reflect-metadata` create side-effects)

**Comment by kamilmysliwiec** [comment]
@sinedied thank you! I left a 1 minor comment.

File: package.json (hunk 1 above)
Comment (sinedied) [reply package.json]: Would it be ok to move it to peerDepenencies then? as when I linked the lib to debug it, it was complaining about `recflect-metadata` being missing, that's why I added it.

Comment (kamilmysliwiec) [reply package.json]: @sinediad `peerDependencies` is fine

Comment (sinedied) [reply package.json]: Done

**Comment by sinedied** [comment]
I rebased my changed and moved `reflect-metadata` to `peerDependencies`, everything should be fine now 😄

**Comment by kamilmysliwiec** [comment]
Thank you :)

**Pull request merged**